SteamKit2/SteamKit2/Util/Win32Helpers.cs

```diff
@@ -165,24 +166,25 @@ protected override bool ReleaseHandle()
 			}
 		}
 
-		static class NativeMethods
+		static partial class NativeMethods
 		{
 			#region CreateFile
 
 			public static uint FILE_SHARE_READ = 1;
 			public static uint FILE_SHARE_WRITE = 2;
 			public static uint OPEN_EXISTING = 3;
 
-			[DllImport( "kernel32.dll", SetLastError = true, CharSet = CharSet.Unicode )]
-			public static extern FileSafeHandle CreateFile( string lpFileName, uint dwDesiredAccess, uint dwShareMode, IntPtr lpSecurityAttributes, uint dwCreationDisposition, uint dwFlagsAndAttributes, IntPtr hTemplateFile );
+			[LibraryImport( "kernel32.dll", EntryPoint = "CreateFileW", SetLastError = true, StringMarshalling = StringMarshalling.Utf16 )]
```

Comment:
The Windows APIs should probably use [CsWin32](https://github.com/microsoft/CsWin32) rather than having us hand-roll the P/Invokes.

Reply:
If only this was a thing for mac

Reply:
I've been poking at the edges of it for macOS and libc but need to figure out how they get the definitions. I might open an issue.